packages/api/src/aggregate/WhereClause.ts

```diff
@@ -195,5 +211,41 @@ export type WhereClause<
   | AndWhereClause<T>
   | NotWhereClause<T>
   | {
-    [P in PropertyKeys<T>]?: FilterFor<CompileTimeMetadata<T>["properties"][P]>;
+    [P in PropertyKeys<T>]?:
+      | RequireExactlyOne<
+        Pick<
+          WhereClauseKeys<T, P>,
+          NonNeverKeys<
+            WhereClauseKeys<T, P>
+          >
+        >
+      >
+      | FilterRawType<CompileTimeMetadata<T>["properties"][P]>;
   };
+
+type WhereClauseKeys<
+  T extends ObjectOrInterfaceDefinition,
+  P extends PropertyKeys<T>,
+> = FilterDollarSignKeys<
+  FlattenUnion<FilterFor<CompileTimeMetadata<T>["properties"][P]>>
+>;
+
+type FlattenUnion<T> = {
```

Comment:
Referred to this https://www.roryba.in/programming/2019/10/12/flattening-typescript-union-types.html#:~:text=unnecessary%20runtime%20cost.-,FlattenUnion,-Now%2C%20the%20main